### PR TITLE
Remove DuckDB Parquet implementation

### LINKED ISSUE #5108: Remove Parquet implementation based on DuckDB
### Description

Now that we have the native implementation for Parquet, we can remove the implementation based on DuckDB.
We must port existing DuckDB tests to Native and verify parity.

### PR DESCRIPTION
**Pull request by isadikov**
The PR removes the DuckDB Parquet implementation so we only use native Velox implementation for reads and writes.

I removed `parquet::ParquetReaderType` as it is not needed anymore. I also updated `velox/dwio/parquet/tests/ParquetTpchTestBase.h`, the test now uses native Parquet writer to prepare tables for the test. 

Closes #5108.

### CODE REVIEW

**Comment by isadikov** [comment]
I am still working on it, had a few issues with Parquet writer setup.

File: velox/dwio/parquet/RegisterParquetReader.h [hunk 1]
```diff
@@ -18,10 +18,10 @@
 
 namespace facebook::velox::parquet {
 
-enum class ParquetReaderType { DUCKDB, NATIVE };
```
Comment (majetideepak) [inline velox/dwio/parquet/RegisterParquetReader.h:21]: We don't need this enum.

File: velox/dwio/parquet/CMakeLists.txt [hunk 2]
```diff
@@ -28,8 +27,8 @@ add_library(velox_dwio_parquet_writer RegisterParquetWriter.cpp)
 
 if(VELOX_ENABLE_PARQUET)
   target_link_libraries(
-    velox_dwio_parquet_reader velox_dwio_duckdb_parquet_reader
-    velox_dwio_native_parquet_reader xsimd)
+    velox_dwio_parquet_reader velox_dwio_native_parquet_reader
+    velox_dwio_arrow_parquet_writer xsimd)
```
Comment (aditi-pandit) [inline velox/dwio/parquet/CMakeLists.txt:31]: Why does the native parquet reader require the dwio_arrow_parquet_writer ?

File: velox/dwio/parquet/tests/ParquetTpchTestBase.h [hunk 3]
```diff
@@ -86,22 +89,77 @@ class ParquetTpchTestBase : public testing::Test {
...
+
+      for (auto tpe : schema->children()) {
+        inputColumns.push_back(
+            std::make_shared<connector::hive::HiveColumnHandle>(
+                tpe->name(),
+                connector::hive::HiveColumnHandle::ColumnType::kRegular,
+                tpe));
+      }
+
+      auto config = std::make_unique<core::MemConfig>();
```
Comment (aditi-pandit) [inline velox/dwio/parquet/tests/ParquetTpchTestBase.h:137]: Do the config and conQueryCtx need to be created per table ? Once for all the tables should suffice I think.

File: velox/dwio/parquet/tests/ParquetTpchTestBase.h [hunk 4]
```diff
@@ -86,22 +89,77 @@ class ParquetTpchTestBase : public testing::Test {
...
+    auto queryString = ::duckdb::TPCHExtension::GetQuery(queryNo);
+    // Output of GetQuery() has a new line and a semicolon. These need to be
+    // removed in order to use the query string in a subquery
+    queryString.pop_back(); // remove new line
+    queryString.pop_back(); // remove semi-colon
+    return queryString;
+  }
+
   /// Write TPC-H tables as a Parquet file to temp directory in hive-style
   /// partition
```
Comment (aditi-pandit) [inline velox/dwio/parquet/tests/ParquetTpchTestBase.h:107]: Add fullstop at the end of the comment.

**Review by aditi-pandit — COMMENTED** [review]
Thanks @isadikov for taking this on.

File: velox/dwio/parquet/RegisterParquetReader.h (hunk 1 above)
Comment (isadikov) [reply velox/dwio/parquet/RegisterParquetReader.h:21]: I removed the enum altogether and updated the methods.

File: velox/dwio/parquet/CMakeLists.txt (hunk 2 above)
Comment (isadikov) [reply velox/dwio/parquet/CMakeLists.txt:31]: I think this was a mistake, I removed `dwio_arrow_parquet_writer`.

File: velox/dwio/parquet/tests/ParquetTpchTestBase.h (hunk 4 above)
Comment (isadikov) [reply velox/dwio/parquet/tests/ParquetTpchTestBase.h]: Thanks. Done!

**Comment by isadikov** [comment]
@aditi-pandit @majetideepak Can you review this PR? Thanks.

**Comment by majetideepak** [comment]
@isadikov We need to port the DuckDB reader tests to the native tests.

**Comment by isadikov** [comment]
Could you elaborate or share an example of such test please? Thank you.
Tests are no longer using the DuckDB Parquet reader, TPC-H tests use the TPC-H connector from DuckDB as the source of truth.

**Comment by majetideepak** [comment]
@isadikov thanks!

**Comment by majetideepak** [comment]
@isadikov Let me know when this is ready for review.

**Comment by isadikov** [comment]
@majetideepak @aditi-pandit  The PR is ready for review, please take a look when you have time, thanks.

**Review by aditi-pandit — COMMENTED** [review]
Thanks @isadikov. Overall looks good.

File: velox/dwio/parquet/tests/ParquetTpchTestBase.h [hunk 5]
```diff
@@ -134,53 +197,43 @@ class ParquetTpchTestBase : public testing::Test {
...
-         TO '{}' (FORMAT 'parquet', CODEC 'ZSTD', ROW_GROUP_SIZE {}))"),
+          R"(SELECT s_suppkey, s_name, s_address, s_nationkey, s_phone,
+         s_acctbal::DOUBLE, s_comment FROM {})"),
       std::make_pair(
           "partsupp",
-          R"(COPY (SELECT ps_partkey, ps_suppkey, ps_availqty,
-         ps_supplycost::DOUBLE as supplycost, ps_comment FROM {})
-         TO '{}' (FORMAT 'parquet', CODEC 'ZSTD', ROW_GROUP_SIZE {}))")};
+          R"(SELECT ps_partkey, ps_suppkey, ps_availqty,
+         ps_supplycost::DOUBLE as supplycost, ps_comment FROM {})")};
```
[the remark this answers is not in the document]
Comment (isadikov) [reply velox/dwio/parquet/tests/ParquetTpchTestBase.h:236]: Oh, that part was for writes to Parquet which is not needed anymore. The original queries were like this: 
```sql
COPY (SELECT ... FROM ...) TO <table name> (FORMAT 'parquet', CODEC 'ZSTD', ROW_GROUP_SIZE {});
```
With this PR, we don't write Parquet with DuckDB anymore, so we just keep the SELECT queries. Those queries are used to write data using the native Parquet writer.

File: velox/dwio/parquet/tests/ParquetTpchTest.cpp [hunk 6]
```diff
@@ -15,110 +15,102 @@
  */
 
 #include "velox/common/base/tests/GTestUtils.h"
-#include "velox/dwio/parquet/RegisterParquetReader.h"
 #include "velox/dwio/parquet/tests/ParquetTpchTestBase.h"
 
-class MultiParquetTpchTest
-    : public ParquetTpchTestBase,
-      public testing::WithParamInterface<ParquetReaderType> {
+class MultiParquetTpchTest : public ParquetTpchTestBase {
```
[the remark this answers is not in the document]
Comment (isadikov) [reply velox/dwio/parquet/tests/ParquetTpchTest.cpp]: This is a great point. I have renamed it, thanks!

**Pull request closed without merging**